src/pages/profile/hooks/useCarousel.ts

```diff
@@ -0,0 +1,89 @@
+import { useRef, useEffect } from 'react';
+
+export const useCarousel = (
+  selectedTabIndex: number,
+  setSelectedTabIndex: (index: number) => void,
+  totalIndex: number
+): [React.RefObject<HTMLDivElement>] => {
+  const MIN_DISTANCE = 50;
+  const carouselRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const carousel = carouselRef.current;
+    carousel.scrollLeft = carousel.clientWidth * selectedTabIndex;
+
+    let startX = 0;
+    let distanceX = 0;
+
+    const handlePointerDown = (event: MouseEvent | TouchEvent) => {
+      if (event instanceof TouchEvent) {
+        startX = event.touches[0].pageX;
+      } else {
+        startX = event.pageX;
+      }
+      carousel.addEventListener('mousemove', handlePointerMove);
+      carousel.addEventListener('touchmove', handlePointerMove);
+    };
+
+    const handlePointerMove = (event: PointerEvent | TouchEvent) => {
+      if (event instanceof TouchEvent) {
+        distanceX = event.touches[0].pageX - startX;
+      } else {
+        distanceX = event.pageX - startX;
+      }
+      if (Math.abs(distanceX) < MIN_DISTANCE) {
+        carousel.scrollLeft =
+          carousel.clientWidth * selectedTabIndex - distanceX;
+      }
+    };
+
+    const handlePointerUp = () => {
+      if (
+        distanceX > 0 &&
+        Math.abs(distanceX) > MIN_DISTANCE &&
+        selectedTabIndex > 0
+      ) {
+        setSelectedTabIndex(selectedTabIndex - 1);
+      } else if (
+        distanceX < 0 &&
+        Math.abs(distanceX) > MIN_DISTANCE &&
+        selectedTabIndex < totalIndex
```

Comment:
개인적으로는 조건들을 따로 함수로 분리해서 어떤 조건인지 의미를 알 수 있는 이름을 붙여주시는게 좋지 않을까합니다!

Reply:
```suggestion
        distanceX > 0 &&
        Math.abs(distanceX) > MIN_DISTANCE &&
        selectedTabIndex > 0
      ) {
        setSelectedTabIndex(selectedTabIndex - 1);
      } else if (
        distanceX < 0 &&
        Math.abs(distanceX) > MIN_DISTANCE &&
        selectedTabIndex < totalIndex - 1
```

selectedTabIndex는 `totalIndex - 1`과 비교하는 조건이어야 마지막 탭에서도 우측으로 탭을 넘길 수 있는 문제가 일어나지 않을 것 같습니다

Reply:
잉 오늘 확인해보니까 잘 되네요? 오히려 -1 조건에서는 마지막 탭으로 절대 갈 수가 없는... ㅋㅋㅋ